packages/lib-classifier/src/store/SubjectStore/ImageAndTextSubject/ImageAndTextSubject.spec.js

```diff
@@ -0,0 +1,154 @@
+import { expect } from 'chai'
+import { when } from 'mobx'
+import nock from 'nock'
+import asyncStates from '@zooniverse/async-states'
+
+import { SubjectFactory, WorkflowFactory } from '@test/factories'
+import mockStore from '@test/mockStore'
+import subjectViewers from '@helpers/subjectViewers'
+
+import ImageAndTextSubject from './ImageAndTextSubject'
+
+describe('Model > ImageAndTextSubject', function () {
+  const subjectSnapshot = SubjectFactory.build({
+    content: 'This is test subject content',
+    contentLoadingState: asyncStates.success,
+    locations: [
+      { 'image/png': 'http://localhost:8080/subjectImage.png' },
+      { 'text/plain': 'http://localhost:8080/subjectContent.txt' }
+    ]
+  })
+
+  const successSubjectSnapshot = SubjectFactory.build({
+    locations: [
+      { 'image/png': 'http://localhost:8080/subjectImage.png' },
+      { 'text/plain': 'http://localhost:8080/success.txt' }
+    ]
+  })
+
+  const invalidLocationsSubjectSnapshot = SubjectFactory.build({
+    locations: [
+      { 'text/plain': 'http://localhost:8080/success.txt' },
+      { 'audio/mpeg': 'http://localhost:8080/example.mp3' }
+    ]
+  })
+
+  const failureSubjectSnapshot = SubjectFactory.build({
+    locations: [
+      { 'image/png': 'http://localhost:8080/subjectImage.png' },
+      { 'text/plain': 'http://localhost:8080/failure.txt' }
+    ]
+  })
+
+  const workflowSnapshot = WorkflowFactory.build()
+  let subject
+
+  before(function () {
+    subject = ImageAndTextSubject.create(subjectSnapshot)
+  })
+
+  it('should exist', function () {
+    expect(ImageAndTextSubject).to.be.ok()
+    expect(ImageAndTextSubject).to.be.an('object')
+  })
+
+  it('should have a `locations` property', function () {
+    expect(subject.locations).to.deep.equal(subjectSnapshot.locations)
+  })
+
+  it('should have two locations', function () {
+    expect(subject.locations).to.have.lengthOf(2)
+  })
+
+  it('should have content as expected', function () {
+    expect(subject.content).to.equal(subjectSnapshot.content)
+  })
+
+  it('should have contentLoadingState as expected', function () {
+    expect(subject.contentLoadingState).to.equal(subjectSnapshot.contentLoadingState)
+  })
+
+  describe('with invalid subject locations', function () {
+    it('should throw an error', function () {
+      expect(() => ImageAndTextSubject.create(invalidLocationsSubjectSnapshot)).to.throw()
+    })
+  })
+
+  describe('with location request response that fails', function () {
+    let subject
```

Comment:
`subject` is already defined in the upper scope. This line should be fine, since you're declaring it again with `let`, but using the same name in nested scopes can lead to bugs, in that the value of `subject` from the upper scope will be passed into this function. It can also make code harder to read.